File: 2024/july/javaScript/06.js
/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
var numWaterBottles = function(numBottles, numExchange) {
    function solve(n ,  d ,  rem){
        let extra = Math.trunc((rem + n) / d);
        if (extra == 0) return 0;
        return extra + solve( extra , d ,(n + rem) % d);
    }
    
    return numBottles + solve(numBottles , numExchange , 0);
};

File: 2024/july/java/06.java
class Solution {
    private int solve(int n , int d , int rem){
        int extra = (rem + n) / d;
        if (extra == 0) return 0;
        return extra + solve( extra , d ,(n + rem) % d);
    }
    public int numWaterBottles(int numBottles, int numExchange) {
        return numBottles + solve(numBottles , numExchange , 0);
    }
}
